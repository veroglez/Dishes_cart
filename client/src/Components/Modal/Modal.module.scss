@import "../../common/style.scss";

.modal {
  background-color: rgba(0, 0, 0, 0.5);
  position: absolute;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;

  .content {
    box-sizing: border-box;
    background-color: white;
    padding: 20px;
    top: 0;
    left: 0;
    position: absolute;
    display: flex;
    justify-content: space-between;
    width: 100%;
    height: 100%;
    flex-direction: column;

    @include tablet {
      max-height: 600px;
      width: 80%;
      flex-direction: row;
      transform: translate(-50%, -50%);
      top: 50%;
      left: 50%;
    }

    .info {
      padding: 12px;
    }

    .products {
      overflow: scroll;

      @include tablet {
        width: 70%;
      }
    }

    .product {
      display: flex;
      text-align: left;
      padding-bottom: 20px;

      &:last-child {
        padding-bottom: 0px;
      }

      p { margin-bottom: 8px }

      img {
        width: 100px;
        height: 100px;
      }
    }

    .summary {
      padding: 20px 0;
      text-align: left;
      
      @include tablet {
        border-left: 1px solid #eeeeee;
        padding: 20px;
        width: 30%;
      }

      p {
        flex: 0 0 50%;
      }
    }

    .close {
      position: absolute;
      top: 12px;
      right: 12px;
      width: 24px;
      cursor: pointer;
    }

    button {
      font-size: 16px;
      padding: 16px 24px;
      width: 100%;
    }
  }
}