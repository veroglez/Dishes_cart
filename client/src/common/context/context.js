import { node } from 'prop-types';
import React, { createContext, Component } from 'react';
import l10n from '../l10n';

const Context = createContext();
const { Provider, Consumer: ConsumerData } = Context;

class ProviderData extends Component {
  static propTypes = {
    children: node,
  };

  static defaultProps = {
    children: undefined,
  };

  state = {
    l10n,
    products: [],
    items: 0,
    total: 0,
  }

  onData = object => this.setState(object);

  render() {
    const {
      onData, props: { children },
    } = this;

    return (
      <Provider value={{ ...this.state, onData }}>
        { children }
      </Provider>
    );
  }
}

export { ConsumerData, ProviderData };