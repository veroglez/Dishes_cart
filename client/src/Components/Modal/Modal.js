import React, { PureComponent } from 'react';
import { func } from 'prop-types';
import { Price } from '..';
import { ConsumerData } from '../../common/context/context';
import assets from '../../common/assets';
import style from './Modal.module.scss';

class Modal extends PureComponent {
  static propTypes = {
    onClose: func,
  };

  static defaultProps = {
    onClose: undefined,
  };

  render() {
    const { onClose } = this.props;

    return (
      <ConsumerData>
        {({ l10n, products, total }) => (
          <div className={style.modal}>
            <div className={style.content}>
              <img className={style.close} src={assets.close} alt="" onClick={() => onClose(false)} />
              <div className={style.products}>
                {products.map((product, index) => (
                  <div className={style.product} key={index.toString()}>
                    <img src={product.image} alt={product.title} />
                    <div className={style.info}>
                      <p>{product.title}</p>
                      <Price price={product.price} />
                      <p>{`${l10n.QUANTITY}: ${product.quantity}`}</p>
                    </div>
                  </div>
                ))}
              </div>
              <div className={style.summary}>
                <p>{l10n.TOTAL.toUpperCase()}</p>
                <Price price={total} />
                <button type="button" disabled>{l10n.FINALIZE}</button>
              </div>
            </div>
          </div>
        )}
      </ConsumerData>
    );
  }
}

export default Modal;
