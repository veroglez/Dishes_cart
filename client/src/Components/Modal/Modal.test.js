import React from 'react';
import renderer from 'react-test-renderer';
import Modal from './Modal';
import { ProviderData } from '../../common/context/context';

const PRODUCTS = [
  { name: 'lorem ipsum 1', image: '/image.png', price: 20 },
  { name: 'lorem ipsum 2', image: '/image.png', price: 30 },
];

const DEFAULTS = {
  l10n: require('../../common/l10n'), // eslint-disable-line
  products: PRODUCTS,
  total: 1000,
};

const HOC = inherit => (
  <ProviderData>
    <Modal {...DEFAULTS} {...inherit} />
  </ProviderData>
);

describe('<Modal>', () => {
  it('render', () => {
    const tree = renderer.create(<HOC />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {onClose}', () => {
    const tree = renderer.create(<HOC onClose={() => {}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
