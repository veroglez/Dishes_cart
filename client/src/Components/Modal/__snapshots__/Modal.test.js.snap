// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal> render 1`] = `
<div
  className="modal"
>
  <div
    className="content"
  >
    <img
      alt=""
      className="close"
      onClick={[Function]}
      src=""
    />
    <div
      className="products"
    />
    <div
      className="summary"
    >
      <p>
        TOTAL
      </p>
      <p
        className="price"
      >
        0 €
      </p>
      <button
        disabled={true}
        type="button"
      >
        Finalizar compra
      </button>
    </div>
  </div>
</div>
`;

exports[`<Modal> when {onClose} 1`] = `
<div
  className="modal"
>
  <div
    className="content"
  >
    <img
      alt=""
      className="close"
      onClick={[Function]}
      src=""
    />
    <div
      className="products"
    />
    <div
      className="summary"
    >
      <p>
        TOTAL
      </p>
      <p
        className="price"
      >
        0 €
      </p>
      <button
        disabled={true}
        type="button"
      >
        Finalizar compra
      </button>
    </div>
  </div>
</div>
`;
